documentation/test-scenario/references/Runner.md

```diff
@@ -0,0 +1,130 @@
+
+  // Create the loader from OAV
+  const loader = TestResourceLoader.create({
+    useJsonParser: false,
+    checkUnderFileRoot: false,
+    fileRoot,
+    swaggerFilePaths,
+  });
+
+  // Load the test scenario file. File list could also be specified in readme.test.md
+  const testDef = await loader.load(
+    "Microsoft.ContainerService/stable/2020-12-01/test-scenarios/containerService.yaml"
+  );
+
+  console.log(testDef.testScenarios[0].steps);
+
+  // Setup initial variable env
+  const env = new VariableEnv();
+  env.setBatch({
+    subscriptionId: "__your_subs_id_",
+    location: "westus",
+    SSH_PUBLIC_KEY: "__public_key_ssh__",
+  });
+
+  // Reference runner implementation in OAV. You need to implement your own runner.
+  const runner = new TestScenarioRunner({
+    jsonLoader: loader.jsonLoader,
+    env,
+    client: new TestScenarioRestClient(getDefaultAzureCredential(), {}),
+  });
+
+  try {
+    for (const scenario of testDef.testScenarios) {
+      await runner.executeScenario(scenario);
+    }
+  } catch (e) {
+    console.log(e.message, e.stack);
+  } finally {
+    console.timeLog("TestLoad");
+    await runner.cleanAllTestScope();
+  }
+```
+
+After the test scenario is loaded, the test step will be slightly different from the file content. Every rest step will have the following resolved fields:
+
+- requestParameters
+  - Type: `object`, map of resolved parameter name and value.
+- responseExpected
+  - Type: `any`
+  - The expected response body from the request.
+
+## Procedure of runner
+
+### Input
+
+Let's assume the following things as input of runner:
+
+- Test scenario definition that loaded via OAV.
+- The scenario id of the test scenario. Test scenario definition file could contains multiple test scenarios, runner need to run one of them.
+- Extra environment variables that required by test scenario, defined in `requiredVariables`.
+
+### Scope
+
+It's the `scope` field defined at top level of test scenario file. Now only `ResourceGroup` is supported, it means that the test scenario will:
+
+- Run under specified resource group.
+- Runner would manage the resource group, it could create the resource group (defined by variable `subscriptionId`, `resourceGroup`, `location`) or use predefined resource group, it could also delete the resource group after the test scenario is done. Runner itself is responsible for managing the resource group, the behavior is not defined by this spec.
+- Runner would run all the arm template deployment under the specified resource group.
+
+The scope is a convention, however it would not be enforced by test scenario. User could override the variable `resourceGroup` in any step to run that step in another resource group for example.
+
+### Variables
+
+See [Variables](./Variables.md) for variable spec. The runner must follow the variable definition in test scenario. Runner do not need to care about the variable conventions as it's already resolved by OAV. The runner must:
```

Comment:
Shall we define a clear&simple naming convention for variables, such as 'only alphabet&number, start with number, length 3~20'? 
This will prevent users to give an arbitrary name that don't meet all variable naming conventions for different target codegen language TS/JAVA/GO/PYTHON...

Reply:
Good point. Will add the contraint.